classes/class-schedule.php

```diff
@@ -44,52 +56,33 @@ public function __construct( $id ) {
 
 		// Verify the schedule id
 		if ( ! is_string( $id ) || ! trim( $id ) ) {
-			throw new Exception( 'Argument 1 for ' . __METHOD__ . ' must be a non empty string' );
+			throw new \Exception( 'Argument 1 for ' . __METHOD__ . ' must be a non empty string' );
 		}
 
-		// Setup HM Backup
-		parent::__construct();
-
 		// Store id for later
 		$this->id = $id;
 
 		// Load the options
 		$this->options = array_filter( (array) get_option( 'hmbkp_schedule_' . $this->get_id() ) );
 
-		// Some properties can be overridden with defines
-		if ( defined( 'HMBKP_ROOT' ) && HMBKP_ROOT ) {
-			$this->set_root( HMBKP_ROOT );
-		}
-
-		if ( defined( 'HMBKP_EXCLUDE' ) && HMBKP_EXCLUDE ) {
-			parent::set_excludes( HMBKP_EXCLUDE, true );
-		}
+		// Setup The Backup class
+		$this->backup = new Backup();
```

Comment:
> De-couple Scheduled_Backup from Backup, the former no longer extends the latter, instead relying on dependancy injection.

As I understand it, dependency injection would require the `BackUp` instance to be passed as a parameter to the constructor

````
public function __construct( Backup $backup ) {
  $this->backup = $backup;
}